airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/commons/util/CompositeIterator.kt

```diff
@@ -65,16 +67,20 @@ internal constructor(
         while (!currentIterator().hasNext()) {
             try {
                 currentIterator().close()
-                emitStartStreamStatus(currentIterator().airbyteStream)
-                StreamStatusUtils.emitCompleteStreamStatus(
-                    airbyteStream,
-                    airbyteStreamStatusConsumer
-                )
+                if (emitStatus) {
```

Comment:
on what scenarios would this flag be false? non certified connectors?

Reply:
By default emitStatus is false to prevent every composite iterator sending these traces (we have sometimes multiple layers of composites, each layer emitting status).
See ln. 37:
```
emitStatus: Boolean = false
```
I don't think anything uses or expect these trace messages coming from CompositeIterator, but in case it's needed I put it as an argument.